File: 백준/Silver/2193. 이친수/이친수.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;

public class Main {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        int n = Integer.parseInt(br.readLine());

        if (n == 1) {
            System.out.println(1);
            return;
        }

        long prevZero = 0;
        long prevOne = 1;

        for (int i = 2; i <= n; i++) {
            long curZero = prevZero + prevOne;
            long curOne = prevZero;

            prevZero = curZero;
            prevOne = curOne;
        }

        System.out.println(prevZero + prevOne);
    }
}

File: 백준/Silver/2193. 이친수/이친수.kt
import java.io.BufferedReader
import java.io.InputStreamReader

fun main() {
    val br = BufferedReader(InputStreamReader(System.`in`))
    val n = br.readLine().toInt()

    if (n == 1) {
        println(1)
        return
    }

    var (prevZero, prevOne) = 0L to 1L

    repeat(n - 1) {
        val curZero = prevZero + prevOne
        val curOne = prevZero

        prevZero = curZero
        prevOne = curOne
    }

    println(prevZero + prevOne)
}
